### PR TITLE
[WIP] Rewrite Code Comments

### PR DESCRIPTION
**Pull request by IAmActuallyCthulhu**
I am rewriting the comments into complete sentences with spaces after `//`.

Currently, I am only looking for comments without a space after `//`, as mentioned in #16521.

I am adding `EXPLAIN_THIS_COMMENT:` to comments that need more information/knowledge **OR** just need to look a little closer at the surrounding code and moved on.

---

The idea for this PR is from https://github.com/godotengine/godot/pull/30981#issuecomment-516626317. For those interested, the regex I am using is below.

```regex
(?<!https:)//(?=[a-zA-Z])(?!/|#)
```

Since there is no way to tell if actual is commented (there should not be an additional space added), I can only reduce the search ignoring the comments followed by `/` and `#`.

### CODE REVIEW

**Comment by aaronfranke** [comment]
Glad this is being done eventually, but after this is merged the question will be, should we enforce this for new contributions as well? And perhaps we should write a C++ style guide that enforces more things beyond just what clang-format fixes.

**Comment by akien-mga** [comment]
I think that's a good idea, but it's a big effort. Since this will touch most of the codebase, I would ask to wait for after 3.2 is released and the `vulkan` branch is merged into `master`, to avoid adding merge conflicts everywhere. Rebasing the `vulkan` branch is already complex enough as is :)

We should also define our comment style and expectations in terms of content, and document it in the style guide. It should then be enforced on new contributions too, and gradually applied to the whole codebase.

> I am adding EXPLAIN_THIS_COMMENT: to comments that need more information/knowledge OR just need to look a little closer at the surrounding code and moved on.

Solving this will be a tremendous amount of work given the number of occurrences you added so far. I support the initiative, better comments would be beneficial to everyone. But this also needs to be preceded by defining and documentation what expectations we have of good comments.

**Review by aaronfranke — APPROVED** [review]

File: core/compressed_translation.h [hunk 1]
```diff
@@ -37,12 +37,12 @@ class PHashTranslation : public Translation {
...
 	GDCLASS(PHashTranslation, Translation);
 
-	//this translation uses a sort of modified perfect hash algorithm
-	//it requires hashing strings twice and then does a binary search,
-	//so it's slower, but at the same time it has an extreemly high chance
-	//of catching untranslated strings
+	// This translation uses a sort of modified perfect hash algorithm
+	//	it requires hashing strings twice and then does a binary search,
+	//	so it's slower. However at the same time it has an extremely high
+	//	chance of catching untranslated strings.
```
[the remark this answers is not in the document]
Comment (akien-mga) [reply core/compressed_translation.h]: No, the tabs looks like they were added by mistake.

**Comment by akien-mga** [comment]
I support the idea, but there are various issues with the current PR and rebasing it would be nearly as much work as redoing it from scratch, so closing.

I'd suggest that guidelines are written for a common comment style and requirements before doing such codebase-wide change. Ideally, we should also have tooling to help enforce the coding style without relying on thorough review of the comment style in every single contribution.

**Pull request closed without merging**